pkg/server/container_status.go

```diff
@@ -43,17 +44,20 @@ func (c *criService) ContainerStatus(ctx context.Context, r *runtime.ContainerSt
 	imageRef := container.ImageRef
 	image, err := c.imageStore.Get(imageRef)
 	if err != nil {
-		return nil, errors.Wrapf(err, "failed to get image %q", imageRef)
-	}
-	repoTags, repoDigests := parseImageReferences(image.References)
-	if len(repoTags) > 0 {
-		// Based on current behavior of dockershim, this field should be
-		// image tag.
-		spec = &runtime.ImageSpec{Image: repoTags[0]}
-	}
-	if len(repoDigests) > 0 {
-		// Based on the CRI definition, this field will be consumed by user.
-		imageRef = repoDigests[0]
+		if err != store.ErrNotExist {
+			return nil, errors.Wrapf(err, "failed to get image %q", imageRef)
+		}
```

Comment:
Let's cache the image.References in meta, we don't want to rely on it sticking around. Don't like the container status results changing / omitting image data cause the image ref has been removed from the store or worse the image has been updated / removed, but not for the container. Or at least we can add a TODO and open an issue to do so.

Reply:
`image.References` is dynamically changed. I don't think we can cache it in container meta.

Reply:
that's sort of my point, if the image that was used "changes" it's not what was used for the snapshot, thus the status information is not correct. I don't want to use the cache for anything but the status output.

image.References is just an array of strings.

Reply:
Are you ok with https://github.com/containerd/cri/issues/990#issuecomment-445323970?

Reply:
yes, see responses on #990 

I still think we should cache the image.references at snapshot

Reply:
nvm I see we do cache the original imageref.. that should be enough

Reply:
Are you OK with the change now? :)